src/llm_openai.jl

```diff
@@ -271,6 +271,34 @@ function OpenAI.create_chat(schema::DatabricksOpenAISchema,
             kwargs...)
     end
 end
+function OpenAI.create_chat(schema::AzureOpenAISchema,
+    api_key::AbstractString,
+    model::AbstractString,
+    conversation;
+    api_version::String = "2023-03-15-preview",
+    http_kwargs::NamedTuple = NamedTuple(),
+    streamcallback::Any = nothing,
+    url::String = "https://<resource-name>.openai.azure.com",
+    kwargs...)
+
+    # Build the corresponding provider object
+    provider = OpenAI.AzureProvider(;
+        api_key = isempty(AZURE_OPENAI_API_KEY) ? api_key : AZURE_OPENAI_API_KEY,
+        base_url = isempty(AZURE_OPENAI_ENDPOINT) ? url : AZURE_OPENAI_ENDPOINT,
+        api_version = api_version
+    )
+    # Override standard OpenAI request endpoint
+    OpenAI.openai_request(
+        "openai/deployments/$model/chat/completions",
+        provider;
+        method = "POST",
+        http_kwargs = http_kwargs,
+        messages = conversation,
+        query = Dict("api-version", provider.api_version),
```

Comment:
What's this kwarg? It seems like a header, not a kwarg?

If you need custom HTTP headers, you'll need to overload more functions (with your own "Provider" as per the example in `examples/` folder).

In addition, I don't think you've tested this version because that's an invalid syntax for a dictionary.

Could you please test the code before you request another review?

Reply:
Btw if you're providing it for the "URL" to be rendered with `?api-version...` at the end, you don't have to --> it happens automatically because of this method for AzureProvider: https://github.com/JuliaML/OpenAI.jl/blob/762daadd3ad35ae1badd47fdc02796ca7bd6c886/src/OpenAI.jl#L64

Reply:
The [`request_body`](https://github.com/JuliaML/OpenAI.jl/blob/main/src/OpenAI.jl#L82) method in `OpenAI.jl` uses `HTTP.request("POST", ....)` instead of `HTTP.post`. For whatever reason, `HTTP.request` removes the query parameters of the URL and makes it necessary to pass them explicitly in the `query` argument.